09_json/n0npax/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	puppy "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy/store"
+)
+
+var (
+	out           io.Writer = os.Stdout
+	args                    = os.Args[1:]
+	puppyFilePath           = kingpin.Flag("data", "path to file with puppies data").Short('d').ExistingFile()
+)
+
+func main() {
+	_, err := kingpin.CommandLine.Parse(args)
+	checkError(err)
+	if *puppyFilePath != "" {
+		puppies := readPuppies(*puppyFilePath)
+		store := store.MemStore{}
+		feedStore(store, puppies)
+	}
+}
+
+func readPuppies(path string) []puppy.Puppy {
+	b, err := ioutil.ReadFile(path)
+	checkError(err)
```

Comment:
refactor to:
* return `error` from this function
* accepts `io.Reader` as an argument type
will be done in `lab10` Do I have to fix this in lab09 too or can it stay as is?

Reply:
What u had was fine, I reckon.